TF_Settings_Web/src/Components/Pages/CameraPage.tsx

```diff
@@ -1,18 +1,240 @@
+    constructor(props: {}) {
+        super(props);
+
+        let state: physicalState = {
+            screenHeight: 0,
+            cameraHeight: 0,
+            cameraLeftToRight: 0,
+            screenTilt: 0,
+            cameraRotation: 0,
+            cameraDistanceFromScreen: 0,
+            physicalConfig: {LeapPositionRelativeToScreenBottomM:{X:0,Y:0,Z:0}, LeapRotationD:{X:0,Y:0,Z:0}, ScreenHeightM:0, ScreenRotationD:0},
+            selectedView: "screenHeight"
+        };
+
+        this.state = state;
+    }
+
+    componentDidMount(): void {
+        if (ConnectionManager.serviceConnection()?.webSocket?.readyState === WebSocket.OPEN) {
+            ConfigurationManager.RequestConfigFileState(this.setStateFromFile.bind(this));
+        } else {
+            ConnectionManager.instance.addEventListener('OnConnected', () => {
+                ConfigurationManager.RequestConfigFileState(this.setStateFromFile.bind(this));
+            });
+        }
+    }
+
+    setStateFromFile(config: ConfigState): void {
+        this.setState(() => ({
+            screenHeight: this.roundToFiveDecimals(config.physical.ScreenHeightM * 100),
+            cameraHeight: this.roundToFiveDecimals(config.physical.LeapPositionRelativeToScreenBottomM.Y * 100),
+            cameraLeftToRight: this.roundToFiveDecimals(config.physical.LeapPositionRelativeToScreenBottomM.Z * 100),
+            cameraDistanceFromScreen: this.roundToFiveDecimals(-config.physical.LeapPositionRelativeToScreenBottomM.Z * 100),
+            cameraRotation: config.physical.LeapRotationD.X,
+            physicalConfig: config.physical,
+            screenTilt: config.physical.ScreenRotationD
+        }));
+    }
+
+    roundToFiveDecimals(numberIn: number): number {
+        return Math.round(numberIn * 100000) / 100000;
+    }
+
+    componentDidUpdate(prevProps: {}, prevState: physicalState): void {
+        if (this.state !== null &&
+            this.state !== prevState) {
+            let physicalConfigState: PhysicalConfig = {
+                ...this.state.physicalConfig,
+                LeapPositionRelativeToScreenBottomM: {
+                    X: this.state.cameraLeftToRight / 100,
+                    Y: this.state.cameraHeight / 100,
+                    Z: -this.state.cameraDistanceFromScreen / 100
+                },
+                LeapRotationD: {
+                    ...this.state.physicalConfig.LeapRotationD,
+                    X: this.state.cameraRotation
+                },
+                ScreenHeightM: this.state.screenHeight / 100,
+                ScreenRotationD: this.state.screenTilt
+            }
+            ConfigurationManager.RequestConfigFileChange(null, physicalConfigState, this.configChangeCbHandler.bind(this));
+        }
+    }
+
+    configChangeCbHandler(result: WebSocketResponse): void {
+        if (result.status !== "Success") {
+            console.error(`Failed to set config state! Info: ${result.message}`);
+        }
+    }
+
```

Comment:
Got some odd whitespace in here, there's two spaces here but then 0 spaces between a bunch of the functions below?